rclcpp/src/rclcpp/executor.cpp

```diff
@@ -623,7 +623,12 @@ Executor::execute_subscription(rclcpp::SubscriptionBase::SharedPtr subscription)
     // Deliver ROS message
     case rclcpp::DeliveredMessageKind::ROS_MESSAGE:
       {
-        if (subscription->can_loan_messages()) {
+        // TODO(clalancette): The loaned message interface is currently not safe to use.
+        // If a user takes a reference to the shared_ptr, it can get freed from underneath them
```

Comment:
Technically it's not a reference to the shared pointer that's the issue. That wouldn't be safe with or without loaning. It's if they copy the shared pointer.